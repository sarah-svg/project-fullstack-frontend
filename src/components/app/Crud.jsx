/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';

function Crud({ onSubmit, onChange, image, locations, id, method }) {
  return (
    <div>
      <form encType="multipart/form-data" action="/" onSubmit={onSubmit}>
        <title>Map</title>
        <h1>Please select one of
                    the options to add, update, or get all of you maps</h1>
        <div>
          <input
            id="post"
            type="radio" 
            name="method" 
            placeholder="Enter post" 
            value="post"
            checked={method === 'post'}
            onChange={onChange}
          />
          <label htmlFor="post">Add a map to your database</label>
          <input
            type="file" placeholder=" add new map image"
            name="image" value={image} onChange={onChange}
          />
          <input
            type="text" placeholder=" add new map locations"
            name="location" value={locations} onChange={onChange}         
          />
        </div>
        <div>
          <input 
            id="get"
            type="radio" 
            name="method" 
            value="get" 
            checked={method === 'get'}
            onChange={onChange} />
          <label htmlFor="get">Get all Maps</label>
        </div>
        <div>
          <input 
            id="find"
            type="radio" 
            name="method" 
            value="find"
            checked={method === 'find'}
            onChange={onChange} />
          <label htmlFor="find">Find Map by Id</label>

          <input 
            type="text" 
            placeholder="Enter ID of Map" 
            name="id"
            value={id}
            onChange={onChange} />
        </div>
              
        <div>
          <input 
            id="delete"
            type="radio" 
            name="method" 
            value="delete"
            checked={method === 'delete'}
            onChange={onChange} />
          <label htmlFor="delete">Delete Map by Id</label>

          <input 
            type="text" 
            placeholder="Enter ID of Map" 
            name="id"
            value={id}
            onChange={onChange} />
        </div>
        <div>
          <input 
            id="put"
            type="radio" 
            name="method" 
            value="put"
            checked={method === 'put'}
            onChange={onChange} />
          <label htmlFor="put">Update map by Id</label>

          <input 
            type="text" 
            placeholder="Enter ID of Map" 
            name="id"
            value={id}
            onChange={onChange} />

          <br />
          <input 
            type="text" 
            placeholder="Update image" 
            name="image"
            value={image}
            onChange={onChange} />
  
          <input 
            type="text" 
            placeholder="{1,2,3}" 
            name="locations"
            value={locations}
            onChange={onChange} />
  
        </div>
        <button>Submit</button>
      </form>
    </div>
  );
}

Crud.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  image: PropTypes.string,
  locations: PropTypes.string,
  method: PropTypes.string,
  id: PropTypes.string
};

export default Crud;

