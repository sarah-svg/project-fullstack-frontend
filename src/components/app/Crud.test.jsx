import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { MemoryRouter } from 'react-router';
import Crud from './Crud';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<MemoryRouter> <Crud/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});