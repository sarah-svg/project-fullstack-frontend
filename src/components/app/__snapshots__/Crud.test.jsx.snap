// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
   
  <div>
    <form>
      <title>
        Map
      </title>
      <h1>
        Please select one of the options to add, update, or get all of you maps
      </h1>
      <div>
        <input
          id="post"
          name="method"
          placeholder="Enter post"
          type="radio"
          value="post"
        />
        <label
          for="post"
        >
          Add a map to your database
        </label>
        <input
          name="image"
          placeholder=" add new map image"
          type="text"
          value=""
        />
        <input
          name="location"
          placeholder=" add new map locations"
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          id="get"
          name="method"
          type="radio"
          value="get"
        />
        <label
          for="get"
        >
          Get all Maps
        </label>
      </div>
      <div>
        <input
          id="find"
          name="method"
          type="radio"
          value="find"
        />
        <label
          for="find"
        >
          Find Map by Id
        </label>
        <input
          name="id"
          placeholder="Enter ID of Map"
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          id="delete"
          name="method"
          type="radio"
          value="delete"
        />
        <label
          for="delete"
        >
          Delete Map by Id
        </label>
        <input
          name="id"
          placeholder="Enter ID of Map"
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          id="put"
          name="method"
          type="radio"
          value="put"
        />
        <label
          for="put"
        >
          Update map by Id
        </label>
        <input
          name="id"
          placeholder="Enter ID of Map"
          type="text"
          value=""
        />
        <br />
        <input
          name="image"
          placeholder="Update image"
          type="text"
          value=""
        />
        <input
          name="locations"
          placeholder="{1,2,3}"
          type="text"
          value=""
        />
      </div>
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
